lib/src/main/kotlin/io/github/compose4gtk/adw/components/NavigationView.kt

```
@@ -0,0 +1,140 @@
+package io.github.compose4gtk.adw.components
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ComposeNode
+import androidx.compose.runtime.remember
+import io.github.compose4gtk.GtkApplier
+import io.github.compose4gtk.GtkComposeWidget
+import io.github.compose4gtk.GtkContainerComposeNode
+import io.github.compose4gtk.modifier.Modifier
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.gnome.gtk.Widget
+import org.gnome.adw.NavigationPage as AdwNavigationPage
+import org.gnome.adw.NavigationView as AdwNavigationView
+
+private val logger = KotlinLogging.logger {}
+
+private class AdwNavigationViewComposeNode(gObject: AdwNavigationView) :
+    GtkContainerComposeNode<AdwNavigationView>(gObject) {
+    override fun addNode(index: Int, child: GtkComposeWidget<Widget>) {
+        if (child.widget is AdwNavigationPage) {
+            when (index) {
+                children.size -> widget.add(child.widget)
+                0 -> widget.insertAfter(child.widget, null)
+                else -> widget.insertBefore(child.widget, children[index - 1])
+            }
+            super.addNode(index, child)
+        } else {
+            logger.warn { "Only navigation pages can be added to a navigation view." }
```

I'd throw an exception here, for the same reason: it's something 100% under the control of the developer